Sources/ApodiniNetworking/HTTP/Headers.swift

```diff
@@ -0,0 +1,498 @@
+        expires: Date?,
+        maxAge: Int?,
+        domain: String?,
+        path: String?,
+        secure: Bool?, // swiftlint:disable:this discouraged_optional_boolean
+        httpOnly: Bool?, // swiftlint:disable:this discouraged_optional_boolean
+        sameSite: SameSite?
+    ) {
+        self.cookieName = cookieName
+        self.cookieValue = cookieValue
+        self.expires = expires
+        self.maxAge = maxAge
+        self.domain = domain
+        self.path = path
+        self.secure = sameSite == SameSite.none ? true : secure
+        self.httpOnly = httpOnly
+        self.sameSite = sameSite
+    }
+    
+    
+    public init?(httpHeaderFieldValue value: String) {
+        fatalError("Not yet implemented")
+    }
+    
+    
+    public func encodeToHTTPHeaderFieldValue() -> String {
+        var retval = "\(cookieName)=\(cookieValue)"
+        if let expires = expires {
+            retval.append("; Expires=\(expires.encodeToHTTPHeaderFieldValue())")
+        }
+        if let maxAge = maxAge {
+            retval.append("; MaxAge=\(maxAge.encodeToHTTPHeaderFieldValue())")
+        }
+        if let domain = domain {
+            retval.append("; Domain=\(domain.encodeToHTTPHeaderFieldValue())")
+        }
+        if let path = path {
+            retval.append("; Path=\(path.encodeToHTTPHeaderFieldValue())")
+        }
+        if let secure = secure, secure {
+            retval.append("; Secure")
+        }
+        if let httpOnly = httpOnly, httpOnly {
+            retval.append("; HttpOnly")
+        }
+        if let sameSite = sameSite {
+            retval.append("; SameSite=\(sameSite.rawValue)")
+        }
+        return retval
+    }
+}
+
+
+public enum AuthorizationHTTPHeaderValue: HTTPHeaderFieldValueCodable {
+    case basic(credentials: String)
+    /// See [RFC6750](https://datatracker.ietf.org/doc/html/rfc6750)
+    case bearer(token: String)
+    case other(type: String, credentials: String)
+    
+    
+    public init?(httpHeaderFieldValue value: String) {
+        guard let typeEndIdx = value.firstIndex(of: " ") else {
+            return nil
+        }
+        let type = String(value[..<typeEndIdx])
+        let rawCredentials = String(value[typeEndIdx...].dropFirst())
+        switch type.lowercased() {
+        case "basic":
+            self = .basic(credentials: rawCredentials)
+        case "bearer":
+            self = .bearer(token: rawCredentials)
+        default:
+            self = .other(type: type, credentials: rawCredentials)
+        }
+    }
+    
+    public func encodeToHTTPHeaderFieldValue() -> String {
+        switch self {
+        case .basic(let credentials):
+            return "Basic \(credentials)"
+        case .bearer(let token):
+            return "Bearer \(token)"
+        case let .other(type, credentials):
+            return "\(type) \(credentials)"
+        }
+    }
+}
+
+
+public enum ETagHTTPHeaderValue: HTTPHeaderFieldValueCodable {
```

Comment:
Those typed implementations of Header values currently duplicated equivalent ones of those implemented via the `Information` system. See `ApodiniHTTPProtocol/Information` (e.g. the `ETag.swift`). Might make sense to rely on a single implementation for those(?).

Reply:
You're right, same also applies to the MediaType stuff, we have (with this PR, but also before) two implementations of that: we were always using Vapor's implementation for things like setting Content-Type headers for encoded output, but then also had a custom implementation which is used exclusively for specifying a `Blob`'s content type.

I'm planning on merging these two, and would then also merge the ETag stuff, although I'd like this to be a separate PR since it'll most likely require making changes to the package structure.

Reply:
I agree with @Supereg here, would be good to merge these implementation. A separate PR is fine with me as well, we should nevertheless document all open issues (including this one) in GitHub Issues so we don't forget about them 👍